docs/advanced.adoc

```diff
@@ -139,6 +139,144 @@ The resolution function will query contents of type post having the current auth
 <5> Adds a new field "posts" returning a list of posts. 
 
 
-These are only examples. You could also modify the type 'Query' and add an entire new API next to the Headless CMS API. 
+These are only examples. You could also modify the type `Query` and add an entire new API next to the Headless CMS API.
+
+
+Starting from image:images/v-5.1.0.svg[Since verision,opts=inline] Guillotine provides two new methods to simplify work with subscriptions and executing of GraphQL query:
+
+- `createWebSocketData(req)` - create data object for websocket from request
+
+[source,javascript]
+----
+webSocket: {
+    data: guillotineLib.createWebSocketData(req),
+    subProtocols: ['graphql-ws']
+}
+----
+
+- `execute(params)` - This method allows you to execute a GraphQL query. To execute this method, will be enough to pass query and variables. In this case, a schema will be created for each branch of the site with the ability to customize it using the schemaOptions parameter. This schema will be automatically updated when the application is added to the site or when it is removed and re-deployed. If a schema is passed that was created outside of this method, then it is the developer's responsibility to update it.
+
+List of properties for `params` object:
+
+|===
+|Name | Description | Default value
+
+|query:String
+|GraphQL query. Property is required.
+|
+
+|variables:Object
+|Variables for GraphQL query. Property is required.
+|
+
+|siteId:String
+|Site ID. Property is optional.
+|ID of a current site.
+
+|branch:String
+|Branch. Property is optional.
+|Branch from request.
+
+|schema:SCHEMA
+|GraphQL SCHEMA. Property is optional.
+|
+
+|schemaOptions:Object
+|SchemaOptions object to customize schema. Property is optional.
+|
+|===
+
+List of properties for `schemaOptions` object:
+
+|===
+|Name | Description
+
+|applications: String or Array.<String>
+|Allowed application keys in addition to this site. Property is optional.
+
+|allowPaths:String or Array.<String>
+|Allowed content paths in addition to this site. Property is required.
```

Comment:
Why is this required? I guess the requirement is to use at least one of the three options?

Reply:
Good catch. It's my mistake. Sure, that option is optional